### PR TITLE
[CAL-1008] Fix 'All Bookings' filter not getting selected after coming from 'Your Bookings'

### LINKED ISSUE #6937: [CAL-1008] Booking filters weird edge case
When selecting your bookings you for some reason cant click all bookings

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-1008](https://linear.app/calcom/issue/CAL-1008/booking-filters-weird-edge-case)</sub>

### PR DESCRIPTION
**Pull request by asadath1395**
## What does this PR do?

[CAL-1008] Fix 'All Bookings' filter not getting selected after coming from 'Your Bookings'

Fixes #6937 

https://github.com/calcom/cal.com/assets/37188392/f7dd447f-cccd-42a1-a94d-3c96d767fc45

**Environment**: Staging(main branch) / Production

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- Navigate to `/bookings/upcoming`
- Check the `All Bookings` filter

### CODE REVIEW

File: packages/lib/hooks/useTypedQuery.ts [hunk 1]
```diff
@@ -86,5 +86,19 @@ export function useTypedQuery<T extends z.AnyZodObject>(schema: T) {
     }
   }
 
-  return { data: parsedQuery, setQuery, removeByKey, pushItemToKey, removeItemByKeyAndValue };
+  // Remove all query params from the URL
+  function removeAllQueryParams(keys: OutputOptionalKeys[]) {
```
Comment (joeauyeung) [inline packages/lib/hooks/useTypedQuery.ts:90]: I noticed there is a non-optional function param but when you call this function, you don't pass any params.

**Review by joeauyeung — COMMENTED** [review]
Works as intended but please see comment

File: packages/lib/hooks/useTypedQuery.ts (hunk 1 above)
Comment (asadath1395) [reply packages/lib/hooks/useTypedQuery.ts]: @joeauyeung Initially i had planned to create a function to remove multiple keys from the query param, then later on decided on the current approach and my bad i forgot to remove the unnecessary param

File: packages/features/bookings/components/TeamFilter.tsx [hunk 2]
```diff
@@ -38,7 +38,7 @@ export const TeamsMemberFilter = () => {
           checked={!query.teamIds && !query.userIds?.includes(session.data?.user.id || 0)}
           onChange={() => {
             setDropdownTitle(t("all_bookings_filter_label"));
-            removeByKey("teamIds"); // Always clear on toggle  or not toggle (seems weird but when you know the behviour it works well )
```
Comment (joeauyeung) [inline packages/features/bookings/components/TeamFilter.tsx:41]: @sean-brydon I saw that you made this comment. Are you ok with this approach?

**Review by joeauyeung — APPROVED** [review]
LGTM and working. Waiting for @sean-brydon's feedback

File: packages/features/bookings/components/TeamFilter.tsx (hunk 2 above)
Comment (sean-brydon) [reply packages/features/bookings/components/TeamFilter.tsx:41]: Yeah works for me :)

**Review by sean-brydon — APPROVED** [review]
Noice 👍

**Pull request merged**